gwcs/tags/wcs.py

```diff
@@ -0,0 +1,305 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+from asdf import yamlutil
+from asdf.tests import helpers
+from ..gwcs_types import GWCSTransformType, GWCSType
+from ..coordinate_frames import *
+from ..wcs import WCS
+
+
+_REQUIRES = ['astropy']
+
+
+__all__ = ["WCSType", "CelestialFrameType", "CompositeFrameType", "FrameType",
+           "SpectralFrameType", "StepType"]
+
+
+class WCSType(GWCSType):
+    name = "wcs"
+    requires = _REQUIRES
+    types = [WCS]
+    version = '1.0.0'
+
+    @classmethod
+    def from_tree(cls, node, ctx):
+
+        steps = [(x['frame'], x.get('transform')) for x in node['steps']]
+        name = node['name']
+
+        return WCS(steps, name=name)
+
+    @classmethod
+    def to_tree(cls, gwcsobj, ctx):
+        def get_frame(frame_name):
+            frame = getattr(gwcsobj, frame_name)
+            if frame is None:
+                return frame_name
+            return frame
+
+        frames = gwcsobj.available_frames
+        steps = []
+        for i in range(len(frames) - 1):
+            frame_name = frames[i]
+            frame = get_frame(frame_name)
+            transform = gwcsobj.get_transform(frames[i], frames[i + 1])
+            steps.append(StepType({'frame': frame, 'transform': transform}))
+        frame_name = frames[-1]
+        frame = get_frame(frame_name)
+        steps.append(StepType({'frame': frame}))
+
+        return {'name': gwcsobj.name,
+                'steps': yamlutil.custom_tree_to_tagged_tree(steps, ctx)}
+
+    @classmethod
+    def assert_equal(cls, old, new):
+        assert old.name == new.name
+        assert len(old.available_frames) == len(new.available_frames)
+        for (old_frame, old_transform), (new_frame, new_transform) in zip(
+                old.pipeline, new.pipeline):
+            helpers.assert_tree_match(old_frame, new_frame)
+            helpers.assert_tree_match(old_transform, new_transform)
+
+
+class StepType(dict, GWCSType):
+    name = "step"
+    requires = _REQUIRES
+    version = '1.0.0'
+
+
+class FrameType(GWCSType):
+    name = "frame"
+    #requires = ['astropy-1.3.3']
```

Comment:
@drdavella Is this requirement still necessary? I'm guessing not b ut want to confirm.

Reply:
No it's not. ASDF now has a hard dependency on `astropy-3.0`, so it doesn't make sense to account for earlier versions anymore.